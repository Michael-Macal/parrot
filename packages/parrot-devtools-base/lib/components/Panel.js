'use strict';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, arguments); return new Promise(function (resolve, reject) { function step(key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { return Promise.resolve(value).then(function (value) { step("next", value); }, function (err) { step("throw", err); }); } } return step("next"); }); }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

;

var Panel = function (_React$Component) {
  _inherits(Panel, _React$Component);

  function Panel() {
    var _ref,
        _this2 = this;

    var _temp, _this, _ret;

    _classCallCheck(this, Panel);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Panel.__proto__ || Object.getPrototypeOf(Panel)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      loading: true,
      scenarios: {},
      dropdownValue: "",
      error: null,
      baseUrl: {}
    }, _this.setScenario = function (_ref2) {
      var dropdownValue = _ref2.value;


      chrome.runtime.sendMessage({ debug: true, dropdownValue: dropdownValue });

      _this.setState({
        dropdownValue: dropdownValue
      }, _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
        var scenarioUrl, resp;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                scenarioUrl = url.format(_extends({}, _this.state.baseUrl, { pathname: 'scenario' }));
                _context.prev = 1;
                _context.next = 4;
                return fetch(scenarioUrl, {
                  method: 'POST',
                  headers: {
                    Accept: 'application/json',
                    'Content-Type': 'application/json'
                  },
                  body: JSON.stringify({
                    scenario: dropdownValue
                  })
                });

              case 4:
                resp = _context.sent;


                if (resp.ok) {
                  chrome.devtools.inspectedWindow.reload(null);
                }
                _context.next = 11;
                break;

              case 8:
                _context.prev = 8;
                _context.t0 = _context['catch'](1);

                _this.setState({
                  error: _context.t0
                });

              case 11:
              case 'end':
                return _context.stop();
            }
          }
        }, _callee, _this2, [[1, 8]]);
      })));
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Panel, [{
    key: 'normalizeScenarios',
    value: function normalizeScenarios(scenarios) {
      var normalized = [];
      for (var key in scenarios) {
        normalized.push({
          label: key,
          value: key
        });
      }

      return normalized;
    }
  }, {
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this3 = this;

      var tabId = chrome.devtools.inspectedWindow.tabId;

      chrome.runtime.sendMessage({ tabId: tabId }, function () {
        var _ref4 = _asyncToGenerator(regeneratorRuntime.mark(function _callee2(res) {
          var tabUrl, baseUrl, scenariosUrl, resp, body, keys;
          return regeneratorRuntime.wrap(function _callee2$(_context2) {
            while (1) {
              switch (_context2.prev = _context2.next) {
                case 0:
                  tabUrl = url.parse(res.url);
                  baseUrl = {
                    host: tabUrl.host,
                    port: tabUrl.port,
                    protocol: tabUrl.protocol
                  };
                  scenariosUrl = url.format(_extends({}, baseUrl, { pathname: 'scenarios' }));


                  chrome.runtime.sendMessage({ debug: true, url: scenariosUrl });

                  _context2.prev = 4;
                  _context2.next = 7;
                  return fetch(scenariosUrl, {
                    headers: {
                      Accept: 'application/json'
                    },
                    mode: 'cors'
                  });

                case 7:
                  resp = _context2.sent;
                  _context2.next = 10;
                  return resp.json();

                case 10:
                  body = _context2.sent;
                  keys = Object.keys(body);


                  _this3.setState({
                    loading: false,
                    scenarios: body,
                    dropdownValue: keys[0],
                    baseUrl: baseUrl
                  });
                  _context2.next = 19;
                  break;

                case 15:
                  _context2.prev = 15;
                  _context2.t0 = _context2['catch'](4);

                  chrome.runtime.sendMessage({ debug: true, error: JSON.stringify(_context2.t0.stack, null, 4) });

                  _this3.setState({
                    error: _context2.t0
                  });

                case 19:
                case 'end':
                  return _context2.stop();
              }
            }
          }, _callee2, _this3, [[4, 15]]);
        }));

        return function (_x) {
          return _ref4.apply(this, arguments);
        };
      }());
    }
  }, {
    key: 'render',
    value: function render() {
      if (this.state.error) {
        return React.createElement(
          'pre',
          null,
          this.state.error.message
        );
      }

      if (this.state.loading) {
        return React.createElement(
          'p',
          null,
          'loading...'
        );
      }

      return React.createElement(
        'div',
        { className: 'pad' },
        React.createElement(Dropdown, {
          id: 'scenarios',
          label: 'Scenarios',
          options: this.normalizeScenarios(this.state.scenarios),
          value: this.state.dropdownValue,
          onChange: this.setScenario
        })
      );
    }
  }]);

  return Panel;
}(React.Component);