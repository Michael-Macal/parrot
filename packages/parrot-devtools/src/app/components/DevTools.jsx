import React, { Component } from 'react';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';
import styled from 'styled-components';

// Material UI Components
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Paper from 'material-ui/Paper';
import CircularProgress from 'material-ui/CircularProgress';

// Plugin Settings Components
import ListenerControls from './plugins/ListenerControls';
import GeneralControls from './plugins/GeneralControls';
import MiddlewareControls from './plugins/MiddlewareControls';

import fetchApi from '../utils/fetchApi';

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin();

const pluginsConfig = {
  'parrot-middleware': {
    component: MiddlewareControls,
    label: 'Middleware Settings',
  },
  'parrot-listener': {
    component: ListenerControls,
    label: 'Listener Settings',
  },
};

const DevSection = styled.div`padding-top: 20px;`;
const DevHeader = styled.h5`margin: 0;`;

const paperStyle = {
  position: 'absolute',
  padding: '0 20',
  top: 10,
  bottom: 10,
  left: 10,
  right: 10,
};

class DevTools extends Component {
  static propTypes = {
    url: PropTypes.string,
  };

  static defaultProps = {
    url: 'http://localhost:3000',
  };

  state = {
    plugins: [],
    url:
      (window && window.localStorage && window.localStorage.getItem('parrotHostname')) ||
      this.props.url,
    error: false,
  };

  componentDidMount() {
    this.loadSections().catch(error => this.setState({ error }));
  }

  setUrl = url => {
    this.setState({ url });
    if (window && window.localStorage) {
      window.localStorage.setItem('parrotHostname', url);
    }
  };

  loadSections = async () => {
    // Empty cache
    this.setState({ error: false, plugins: [] });
    try {
      const resp = await fetchApi(this.state.url, '/parrot/registry');
      const registry = await resp.json();
      if (registry && registry.middlewares) {
        this.setState({ plugins: registry.middlewares });
      }
    } catch (error) {
      this.setState({ error });
    }
  };

  render() {
    let pluginSections;
    if (this.state.error) {
      pluginSections = (
        <DevSection>
          <h4>Squawk! Error alert!</h4>
          <p>
            Cannot find any Parrot plugins registered, can you check that Parrot is running on that
            hostname?
          </p>
        </DevSection>
      );
    } else if (this.state.plugins.length > 0) {
      pluginSections = this.state.plugins.map(pluginKey => pluginsConfig[pluginKey] || {}).map(
        ({ label, component: SectionComponent }) =>
          label && SectionComponent ? (
            <div>
              <DevSection>
                <DevHeader>{label}</DevHeader>
                <SectionComponent url={this.state.url} />
              </DevSection>
            </div>
          ) : null
      );
    } else {
      pluginSections = (
        <DevSection>
          <CircularProgress />
        </DevSection>
      );
    }

    return (
      <MuiThemeProvider>
        <Paper style={paperStyle}>
          {pluginSections}
          {/* General Settings are always shown */}
          <DevSection>
            <DevHeader>Parrot Settings</DevHeader>
            <GeneralControls
              url={this.state.url}
              setUrl={this.setUrl}
              onRefresh={this.loadSections}
            />
          </DevSection>
        </Paper>
      </MuiThemeProvider>
    );
  }
}

export default DevTools;
