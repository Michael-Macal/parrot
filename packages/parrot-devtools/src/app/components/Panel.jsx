import React, { Component, PropTypes } from 'react';
import url from 'url';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';
import Paper from 'material-ui/Paper';
import CircularProgress from 'material-ui/CircularProgress';

const paperStyle = {
  position: 'absolute',
  top: 10,
  bottom: 10,
  left: 10,
  right: 10,
};

class Panel extends Component {
  static propTypes = {
    url: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
  }

  static defaultProps = {
    url: 'http://localhost:3000',
  }

  state = {
    loading: true,
    scenarios: {},
    dropdownValue: '',
    error: null,
  };

  async componentDidMount() {
    this.baseUrl = url.parse(this.props.url);
    const scenariosUrl = url.format({ ...this.baseUrl, pathname: '/scenarios' });

    try {
      const resp = await fetch(scenariosUrl, {
        headers: {
          Accept: 'application/json',
        },
      });

      const scenarios = await resp.json();
      const keys = Object.keys(scenarios);

      this.setState({ // eslint-disable-line react/no-did-mount-set-state
        loading: false,
        scenarios,
        dropdownValue: keys[0],
      });
    } catch (error) {
      this.setState({ error }); // eslint-disable-line react/no-did-mount-set-state
    }
  }

  setScenario = ({ value: dropdownValue }) => {
    this.setState({
      dropdownValue,
    }, async () => {
      const scenarioUrl = url.format({ ...this.baseUrl, pathname: '/scenario' });

      try {
        const resp = await fetch(scenarioUrl, {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            scenario: dropdownValue,
          }),
        });

        // Reload if we are in chrome
        if (chrome && resp.ok) {
          chrome.devtools.inspectedWindow.reload(null);
        }
      } catch (error) {
        this.setState({ error });
      }
    });
  }

  getContent() {
    if (this.state.error) {
      return <pre>{this.state.error.message}</pre>;
    } else if (this.state.loading) {
      return <CircularProgress />;
    }

    return (
      <DropDownMenu
        value={this.state.dropdownValue}
        onChange={this.setScenario}
      >
        {Object.keys(this.state.scenarios).map(scenario =>
          <MenuItem value={scenario} primaryText={scenario} />,
        )}
      </DropDownMenu>
    );
  }

  render() {
    return (
      <Paper style={paperStyle}>
        {this.getContent()}
      </Paper>
    );
  }
}

export default Panel;
