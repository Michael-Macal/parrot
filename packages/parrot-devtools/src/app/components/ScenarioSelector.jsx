import React, { Component, PropTypes } from 'react';
import url from 'url';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import CircularProgress from 'material-ui/CircularProgress';

class ScenarioSelector extends Component {
  static propTypes = {
    url: PropTypes.string.isRequired, // eslint-disable-line react/no-unused-prop-types
  }

  state = {
    loading: true,
    scenarios: {},
    scenario: '',
    error: null,
  };

  async componentDidMount() {
    try {
      const resp = await fetch(this.createUrl('/scenarios'), {
        headers: {
          Accept: 'application/json',
        },
      });

      const scenarios = await resp.json();
      const keys = Object.keys(scenarios);

      this.setState({ // eslint-disable-line react/no-did-mount-set-state
        loading: false,
        scenarios,
        scenario: keys[0],
      });
    } catch (error) {
      this.setState({ error }); // eslint-disable-line react/no-did-mount-set-state
    }
  }

  setScenario = (event, key, scenario) => {
    this.setState({
      scenario,
    }, async () => {
      try {
        const resp = await fetch(this.createUrl('/scenario'), {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            scenario,
          }),
        });

        if (!resp.ok) {
          this.setState({ error: resp.statusText });
        } else if (chrome) { // Reload if we are in chrome
          chrome.devtools.inspectedWindow.reload(null);
        }
      } catch ({ message: error }) {
        this.setState({ error });
      }
    });
  }

  createUrl = pathname => url.format({ ...url.parse(this.props.url), pathname })

  render() {
    if (this.state.error) {
      return <pre>{this.state.error}</pre>;
    } else if (this.state.loading) {
      return <CircularProgress />;
    }

    return (
      <SelectField
        floatingLabelText="Scenario"
        value={this.state.scenario}
        onChange={this.setScenario}
      >
        {Object.keys(this.state.scenarios).map(scenario =>
          <MenuItem key={scenario} value={scenario} primaryText={scenario} />,
        )}
      </SelectField>
    );
  }
}

export default ScenarioSelector;
