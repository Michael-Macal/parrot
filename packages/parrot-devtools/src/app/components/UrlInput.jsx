import React, { Component, PropTypes } from 'react';
import TextField from 'material-ui/TextField';

const textareaStyle = {
  paddingLeft: 0,
};

class UrlInput extends Component {
  static propTypes = {
    url: PropTypes.string.isRequired,
    setUrl: PropTypes.func.isRequired,
  }

  setUrl = (event, url) => this.props.setUrl(url);

  render() {
    return (
      <TextField
        name="Scenarios Host"
        floatingLabelText="Scenarios Host"
        defaultValue={this.props.url}
        onChange={this.setUrl}
        textareaStyle={textareaStyle}
      />
    );
  }
}

export default UrlInput;
