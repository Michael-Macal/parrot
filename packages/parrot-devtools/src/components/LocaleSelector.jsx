import React, { PropTypes } from 'react';
import { trigger } from 'redial';
import { connect } from 'react-redux';
import { Dropdown } from 'axp-base';
import { UPDATE_LOCALE } from 'axp-global-ducks';

export const LocaleSelector = props => {
  const changeLocale = ({ value }) => {
    props.updateLocale(props.component, value);
  };
  const options = props.availableLocales.map(locale => {
    return {
      value: locale,
      label: locale
    };
  });
  return (
    <Dropdown
      id="localeDropdown"
      options={options}
      onChange={changeLocale}
      value={props.activeLocale}
      title="Choose a locale"
      label="Choose a locale"
    />
  );
};

LocaleSelector.propTypes = {
  activeLocale: PropTypes.string,
  component: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func
  ]).isRequired,
  defaultLocale: PropTypes.string
};

LocaleSelector.defaultProps = {
  activeLocale: 'N/A',
  availableLocales: ['en-US', 'en-NZ']
};

export function mapStateToProps(state) {
  return {
    activeLocale: state.intl.get('activeLocale')
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    updateLocale: (component, locale) => {
      trigger('lang', component, { dispatch, locale }).then(() => {
        dispatch({
          type: 'global/intl/UPDATE_LOCALE',
          locale
        })
      })
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LocaleSelector)
