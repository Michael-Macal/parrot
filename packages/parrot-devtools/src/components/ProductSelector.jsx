import { connect } from 'react-redux';
import React, { PropTypes } from 'react';
import { selectProduct } from 'axp-api-ducks';
import { Dropdown } from 'axp-base';
import { Map } from 'immutable';

const ProductSelector = ({ products, active, select }) => {
  const dropDownProps = {
    id: 'product-selector',
    options: products,
    title: 'Select a product',
    onChange: select,
    value: active,
    label: 'Select a Product'
  };

  return <Dropdown {...dropDownProps} />;
};

ProductSelector.propTypes = {
  products: PropTypes.array,
  active: PropTypes.string,
  select: PropTypes.func,
  onChange: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  const products = Map(state.core.resources.products.get('productsList')).map(product => {
    return {
      text: `${product.account.relationship} - ${product.product.payment_type} - ${product.product.line_of_business_type} - ${product.account_token}`,
      value: product.account_token
    };
  }).toList().toJS();

  
  return {
    products,
    active: state.core.resources.products.get('selectedProduct')
  };
}

function mD2P(dispatch, ownProps) {
  return {
    select: cardId => {
      dispatch(selectProduct(cardId.value))
      ownProps.onChange();
    }
  };
}

export default connect(mapStateToProps, mD2P)(ProductSelector);
