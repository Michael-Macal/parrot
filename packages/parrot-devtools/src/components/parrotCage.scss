.parrot-cage .select[data-rendered]:before {
  white-space: nowrap;
  overflow: hidden;
  width: 100%;
  text-overflow: ellipsis;
  margin: 0 -.625rem;
  padding: 0 1.875rem 0 .625rem;
}