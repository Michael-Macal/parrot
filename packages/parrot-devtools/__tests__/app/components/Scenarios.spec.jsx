/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { ScenariosMiddleware, mapStateToProps } from '../../../src/app/components/Scenarios';
import useScenarios from '../../../src/app/hooks/useScenarios';

jest.mock('../../../src/app/hooks/useScenarios', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    loading: false,
    scenario: '',
    scenarios: [],
    setScenario: jest.fn(() => Promise.resolve()),
    loadScenarios: jest.fn(() => Promise.resolve()),
  })),
}));

describe('Scenarios', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should mapStateToProps', () => {
    expect(mapStateToProps({ url: 'squawk' })).toMatchObject({ url: 'squawk' });
  });

  describe('ScenariosMiddleware', () => {
    const url = 'https://dev.example.com/squawk';

    it('should render scenarios middleware component', () => {
      const scenario = 'api-fetch';
      const setScenario = jest.fn(() => Promise.resolve());

      useScenarios.mockImplementationOnce(() => ({
        scenario,
        setScenario,
        loading: false,
        scenarios: [{ name: scenario }],
        loadScenarios: jest.fn(() => Promise.resolve()),
      }));

      const result = shallow(<ScenariosMiddleware url={url} />);

      expect(useScenarios).toHaveBeenCalledTimes(1);
      expect(useScenarios).toHaveBeenCalledWith(url);
      expect(result).toMatchSnapshot();

      result.find('button').simulate('click');
      expect(setScenario).toHaveBeenCalledWith(scenario);
    });

    it('should loading state', () => {
      useScenarios.mockImplementationOnce(() => ({
        loading: true,
      }));

      const result = shallow(<ScenariosMiddleware url={url} />);

      expect(useScenarios).toHaveBeenCalledTimes(1);
      expect(useScenarios).toHaveBeenCalledWith(url);
      expect(result).toMatchSnapshot();
    });

    it('should inform the user of failure', () => {
      useScenarios.mockImplementationOnce(() => ({
        loading: false,
        scenario: '',
        scenarios: [],
      }));

      const result = shallow(<ScenariosMiddleware url={url} />);

      expect(useScenarios).toHaveBeenCalledTimes(1);
      expect(useScenarios).toHaveBeenCalledWith(url);
      expect(result).toMatchSnapshot();
    });
  });
});
