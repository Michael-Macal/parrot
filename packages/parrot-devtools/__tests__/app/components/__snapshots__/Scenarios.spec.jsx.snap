// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenarios ScenariosMiddleware should inform the user of failure 1`] = `
<styled.div>
  <div
    className="alert alert-warn alert-dialog alert-lg"
    role="alert"
  >
    <span
      className="alert-icon dls-icon-warning icon-lg"
    />
    <h1
      className="heading-4 alert-header"
    >
      Failed to Load Scenarios
    </h1>
    <p
      className="alert-subtitle"
    >
      <span>
        Try 
      </span>
      <button
        type="button"
      >
        refreshing
      </button>
      <span>
         or updating the middleware URL in settings.
      </span>
    </p>
  </div>
</styled.div>
`;

exports[`Scenarios ScenariosMiddleware should loading state 1`] = `
<styled.div>
  <div
    className="progress-circle progress-indeterminate margin-auto"
  />
</styled.div>
`;

exports[`Scenarios ScenariosMiddleware should render scenarios middleware component 1`] = `
<styled.div
  className="border"
>
  <ul
    className="nav-menu"
  >
    <li
      className="nav-item"
      key="api-fetch"
    >
      <button
        aria-pressed={true}
        className="nav-link fluid"
        onClick={[Function]}
        type="button"
      >
        api-fetch
      </button>
    </li>
  </ul>
</styled.div>
`;
